src/NBitcoin/Network.cs

```diff
@@ -239,16 +239,10 @@ public byte[] MagicBytes
         /// </code>
         /// </example>
         /// <returns>A genesis block.</returns>
-        public static Block MineGenesisBlock(ConsensusFactory consensusFactory, string coinbaseText, Target target, Money genesisReward, int version = 1)
+        public static Block MineGenesisBlock(Network network, string coinbaseText, Target target, Money genesisReward, int version = 1)
```

Comment:
Does this method really need to change?

Reply:
It actually appears that we only use this method in tests? Can I just maybe delete it? If not I'll revert some of the changes 👍

Reply:
no it's used for people to generate an actual Genesis block!
so it'll be used by federated sidechains, or anyone wanting to create a genesis block in order to bootstrap a network.

Reply:
OK will change back